controllers/cloudstackmachine_controller.go

```diff
@@ -203,8 +209,20 @@ func (r *CloudStackMachineReconciler) reconcile(
 
 // Reconcile/Destroy a deleted VM instance.
 func (r *CloudStackMachineReconciler) reconcileDelete(
+	ctx context.Context,
 	log logr.Logger,
-	csMachine *infrav1.CloudStackMachine) (ctrl.Result, error) {
+	csMachine *infrav1.CloudStackMachine,
+	capiMachine *capiv1.Machine,
+	csCluster *infrav1.CloudStackCluster) (ctrl.Result, error) {
+
+	// Remove any CAPC managed Affinity groups if owner references a deleted object.
+	if deleted, err := csCtrlrUtils.IsOwnerDeleted(ctx, r.Client, capiMachine); err != nil {
+		return ctrl.Result{}, err
+	} else if deleted {
+		if err := r.RemoveManagedAffinity(log, capiMachine, csMachine, csCluster); err != nil {
+			return ctrl.Result{}, err
```

Comment:
do we want to add any additional explanation of where the error happened and what might've occurred, instead of just returning `err`?

Reply:
Sure, done.